One-exchange neighborhood may raise `ValueError` non-deterministically

# Description
`ConfigSpace.util.get_one_exchange_neighbourhood` uses `np.random` to decide whether a new configuration is to be validated (see [util.pyx:218](https://github.com/automl/ConfigSpace/blob/9fd85c4e7278a471311bd7172e1910bbbe2e850a/ConfigSpace/util.pyx#L218)). Using `np.random` for this decision makes the behavior of the function non-deterministic.

# Solution proposal
Use the randomness generator `random` instantiated at [util.pyx:130](https://github.com/automl/ConfigSpace/blob/9fd85c4e7278a471311bd7172e1910bbbe2e850a/ConfigSpace/util.pyx#L130) instead of `np.random`.

# Minimal example
Try to run the following PCS with `test_sample_configuration_spaces.py`: [vampire.zip](https://github.com/automl/ConfigSpace/files/7170921/vampire.zip)

If all the calls to `cs._check_configuration_rigorous` are removed from the test body, a `ValueError` is raised from `n.is_valid_configuration()` at "test_sample_configuration_spaces.py:74":
```
E   ValueError: Active hyperparameter 'age_weight_ratio:log_ratio' not specified!
```

Note that if the calls to `cs._check_configuration_rigorous` are left in place, a different `ValueError` is raised from `cs._check_configuration_rigorous(default)` at "test_sample_configuration_spaces.py:52":
```
E   ValueError: Inactive hyperparameter 'age_weight_ratio:log_ratio' must not be specified, but has the vector value: '0.7692307692307693'.
```

--- a/ConfigSpace/util.pyx
+++ b/ConfigSpace/util.pyx
@@ -215,7 +215,7 @@ def get_one_exchange_neighbourhood(
                     # Only rigorously check every tenth configuration (
                     # because moving around in the neighborhood should
                     # just work!)
-                    if np.random.random() > 0.95:
+                    if random.random() > 0.95:
                         new_configuration.is_valid_configuration()
                     else:
                         configuration_space._check_forbidden(new_array)

--- a/test/test_util.py
+++ b/test/test_util.py
@@ -103,14 +103,14 @@ class UtilTest(unittest.TestCase):
         hp = UniformFloatHyperparameter('a', 1, 10)
         all_neighbors = self._test_get_one_exchange_neighbourhood(hp)
         all_neighbors = [neighbor['a'] for neighbor in all_neighbors]
-        self.assertAlmostEqual(5.44, np.mean(all_neighbors), places=2)
-        self.assertAlmostEqual(3.065, np.var(all_neighbors), places=2)
+        self.assertAlmostEqual(5.49, np.mean(all_neighbors), places=2)
+        self.assertAlmostEqual(3.192, np.var(all_neighbors), places=2)
         hp = UniformFloatHyperparameter('a', 1, 10, log=True)
         all_neighbors = self._test_get_one_exchange_neighbourhood(hp)
         all_neighbors = [neighbor['a'] for neighbor in all_neighbors]
         # Default value is 3.16
-        self.assertAlmostEqual(3.45, np.mean(all_neighbors), places=2)
-        self.assertAlmostEqual(2.67, np.var(all_neighbors), places=2)
+        self.assertAlmostEqual(3.50, np.mean(all_neighbors), places=2)
+        self.assertAlmostEqual(2.79, np.var(all_neighbors), places=2)
 
     def test_random_neighbor_int(self):
         hp = UniformIntegerHyperparameter('a', 1, 10)

